tests/test_tarfs.py

```diff
@@ -172,3 +172,44 @@ def test_getinfo(self):
         self.assertEqual(info.name, "foo")
         self.assertEqual(info.size, 0)
         self.assertIs(info.type, ResourceType.directory)
+
+
+class TestTarFSReadFromTarCFDotSlashName(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.tarfs = fs.archive.tarfs.TarReadFS(io.BytesIO(cls._build_fixture()))
+
+    @staticmethod
+    def _build_fixture():
+        with fs.memoryfs.MemoryFS() as tmpfs:
+            tmpfile = tmpfs.open("test.tar", "wb+")
+            with tarfile.open(mode="w", fileobj=tmpfile) as tf:
+                tf.addfile(tarfile.TarInfo("./file1"), io.StringIO())
+                info = tarfile.TarInfo("./sub")
+                info.type = tarfile.DIRTYPE
+                tf.addfile(info, io.BytesIO())
+                tf.addfile(tarfile.TarInfo("./sub/file2"), io.StringIO())
```

Comment:
`TarFile.addfile` doesn't need the second argument when it's an empty buffer, so you can just do:
```python
tf.addfile(tarfile.TarInfo("./file1"))
```

Reply:
Fixed